README.md

```diff
@@ -63,6 +65,182 @@ aws_secret_access_key = SECRETKEYHERE
 
 And running the command `kubectl create secret generic aws-creds -n my-namespace --from-file=creds=./creds.conf`
 
+### Using IAM role assumption in EKS
+
+Set environment variables to use with commands below
+```
+export AWS_REGION=region-here
+export CLUSTER_NAME=clustername
+export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query "Account" --output text)
+export IAM_ROLE_NAME=name-to-give-iam-role
+SERVICE_ACCOUNT_NAMESPACE=namespace-ie-crossplane-system-or-upbound-system
+```
+
+The first requirement is that you enable IAM roles for service accounts in your cluster by [Creating an IAM OIDC provider for your cluster](https://docs.aws.amazon.com/eks/latest/userguide/enable-iam-roles-for-service-accounts.html)
+
+```
+eksctl utils associate-iam-oidc-provider --cluster "${CLUSTER_NAME}" --region "${AWS_REGION}" --approve
+```
+
+Our next step is to install Crossplane
+
+```
+kubectl create namespace $SERVICE_ACCOUNT_NAMESPACE
+helm repo add crossplane-stable https://charts.crossplane.io/stable
+helm install crossplane --namespace $SERVICE_ACCOUNT_NAMESPACE crossplane-stable/crossplane
+```
+
+Then install the AWS provider referencing a ControllerConfig containing an annotation
+
+```
+cat > provider.yaml <<EOF
+apiVersion: pkg.crossplane.io/v1alpha1
+kind: ControllerConfig
+metadata:
+  name: aws-config
+  annotations:
+    eks.amazonaws.com/role-arn: arn:aws:iam::$AWS_ACCOUNT_ID:role/$IAM_ROLE_NAME
+spec:
+  podSecurityContext:
+    fsGroup: 2000
+---
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: provider-aws
+spec:
+  package: crossplane/provider-aws:alpha
+  controllerConfigRef:
+    name: aws-config
+EOF
+
+kubectl apply -f provider.yaml
+```
+
+Then we need to create an IAM role for our service account in our EKS cluster's AWS account
+
+```
+kubectl get serviceaccounts -n $SERVICE_ACCOUNT_NAMESPACE
+SERVICE_ACCOUNT_NAME=provider-aws-[id from aws service account name here]
+```
+
+Using EKSCTL
```

Comment:
Does this have the same effect as the "Using AWS CLI" one? with the same condition?

Reply:
Yes, but using EKSCTL is more specific. If we use IAM, then we use StringLike and a wilcard but the EKSCTL method will use StringEquals and the exact name of the AWS Provider service account

I will add it as a note